src/main/java/uk/gov/hmcts/ccd/domain/service/caselinking/CaseLinkService.java

```diff
@@ -82,4 +92,30 @@ private CaseLink setCaseLinkReferences(Long caseReference, CaseLink caseLink) {
         return caseLink;
     }
 
+    private void lockCaseDataRows(Long caseReference, List<CaseLink> caseLinksWithReferences) {
+        List<Long> referencesToLock = new ArrayList<>();
+        referencesToLock.add(caseReference);
+        if (caseLinksWithReferences != null) {
+            referencesToLock.addAll(caseLinksWithReferences.stream()
+                .filter(Objects::nonNull)
+                .map(CaseLink::getLinkedCaseReference)
+                .filter(Objects::nonNull)
+                .distinct()
+                .toList());
+        }
+        if (referencesToLock.isEmpty()) {
+            return;
+        }
+
+        referencesToLock = referencesToLock.stream().sorted().toList();
```

Comment:
this can be removed as order by reference is done on database, if you want to keep this, then order by reference is redundent